infra/build/functions/sync/main.py

```diff
@@ -35,41 +46,138 @@ class GitAuth(ndb.Model):
   access_token = ndb.StringProperty()
 
 
-def sync_projects(projects):
+def create_scheduler(cloud_scheduler_client, project_name, schedule):
+  """Creates schedulers for new projects."""
+  project_id = os.environ.get('GCP_PROJECT')
+  location_id = os.environ.get('FUNCTION_REGION')
+  parent = cloud_scheduler_client.location_path(project_id, location_id)
+  job = {
+      'name': parent + '/jobs/' + project_name + '-scheduler',
+      'pubsub_target': {
+          'topic_name': 'projects/' + project_id + '/topics/request-build',
+          'data': project_name.encode(encoding='UTF-8')
+      },
+      'schedule': schedule
+  }
+
+  cloud_scheduler_client.create_job(parent, job)
+
+
+def delete_scheduler(cloud_scheduler_client, project_name):
+  """Deletes schedulers for projects that were removed."""
+  project_id = os.environ.get('GCP_PROJECT')
+  location_id = os.environ.get('FUNCTION_REGION')
+  name = cloud_scheduler_client.job_path(project_id, location_id,
+                                         project_name + '-scheduler')
+  cloud_scheduler_client.delete_job(name)
+
+
+def update_scheduler(cloud_scheduler_client, project, schedule):
+  """Updates schedule in case schedule was changed."""
+  project_id = os.environ.get('GCP_PROJECT')
+  location_id = os.environ.get('FUNCTION_REGION')
+  parent = cloud_scheduler_client.location_path(project_id, location_id)
+  job = {
+      'name': parent + '/jobs/' + project.name + '-scheduler',
+      'pubsub_target': {
+          'topic_name': 'projects/' + project_id + '/topics/request-build',
+          'data': project.name.encode(encoding='UTF-8')
+      },
+      'schedule': project.schedule
+  }
+
+  update_mask = {'schedule': schedule}
+  cloud_scheduler_client.update(job, update_mask)
+
+
+def sync_projects(cloud_scheduler_client, projects):
   """Sync projects with cloud datastore."""
   project_query = Project.query()
-  projects_to_remove = [
-      project.key for project in project_query if project.name not in projects
-  ]
-
-  ndb.delete_multi(projects_to_remove)
+  for project in project_query:
+    if project.name not in projects:
+      try:
+        delete_scheduler(cloud_scheduler_client, project.name)
+        project.key.delete()
+      except exceptions.GoogleAPICallError as error:
+        logging.error('Scheduler deletion for %s failed with %s', project.name,
+                      error)
 
   existing_projects = {project.name for project in project_query}
 
-  new_projects = [
-      Project(name=project)
-      for project in projects
-      if project not in existing_projects
-  ]
-  ndb.put_multi(new_projects)
+  for project_name in projects:
+    if project_name not in existing_projects:
```

Comment:
bail out early by inverting this condition to avoid the indentation. same with all other cases.